bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemRegistryImpl.java

```diff
@@ -388,31 +388,14 @@ protected void activate(final ComponentContext componentContext) {
     protected void deactivate() {
         super.deactivate();
     }
-
-    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    protected void addStateDescriptionProvider(StateDescriptionProvider provider) {
-        synchronized (stateDescriptionProviders) {
-            stateDescriptionProviders.add(provider);
-
-            Collections.sort(stateDescriptionProviders, new Comparator<StateDescriptionProvider>() {
-                // sort providers by service ranking in a descending order
-                @Override
-                public int compare(StateDescriptionProvider provider1, StateDescriptionProvider provider2) {
-                    return provider2.getRank().compareTo(provider1.getRank());
-                }
-            });
-
-            for (Item item : getItems()) {
-                ((GenericItem) item).setStateDescriptionProviders(stateDescriptionProviders);
-            }
-        }
+    
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.DYNAMIC)
+    protected void addStateDescriptionMergingService(StateDescriptionService stateDescriptionService) {
```

Comment:
since we now have only one service which is injected & the service name is `StateDescriptionService` please rename the method to `setStateDescriptionService`.